TOKEN_GUIDE.md

```diff
@@ -0,0 +1,132 @@
+* "Access-Control-Allow-Methods": "GET, OPTIONS" specifies that only GET and OPTIONS methods are permitted, controlling how other sites can interact with the server.
+
+These headers will allow our API to work well in production and development, but for production you should limit only the origins that should be allowed.
+
+## Input Checks
+To generate the token, we need a few input parameters that we retrieve from the route. We need to check that `channel` and `uid` are not empty. If they are, return a Bad Request Error.
+
+The role determines what type of access the user requesting a token should have. There are two possible roles:
+* publisher - Can publish (or send) their data to the channel. They can also receive data from the channel.
+* subscriber - Can only receive data and cannot publish data.
+
+Those are the only two valid inputs. If one of those inputs is in the URL, we can assign the `role` variable to a number predefined within the `agora-token` package. If the request does not include one of these options, we send back a Bad Request Error.
+
+```ts
+//check for valid channel
+if (!params.channel) {
+    return new Response("channel is required", { status: 400, headers })
+}
+//check for valid role
+let role;
+if (params.role === 'publisher') {
+    role = agoraToken.RtcRole.PUBLISHER;
+} else if (params.role === 'subscriber') {
+    role = agoraToken.RtcRole.SUBSCRIBER
+} else {
+    return new Response("role is incorrect", { status: 400, headers })
+}
+//check for valid uid
+if (!params.uid || params.uid === '') {
+    return new Response("uid is required", { status: 400, headers })
+}
+```
+
+## Generate Token
+At this point, we have all the data needed to generate and return the token. The only values that we haven't defined are the token expiration time and the privilege expiration time. These are in seconds, so I will set them both to 10 minutes (600 seconds).
+
+The privilege expiration time defines how long the user will have their privileges within the channel and access to the channel in general. If the privilege time expires, the user can still listen to the call but can no longer publish. And once the token expires in general, they will lose access to the channel completely.
+
+```ts
+const expireTime = 600;
+const privilegeExpireTime = 600;
+let token;
+
+token = agoraToken.RtcTokenBuilder.buildTokenWithUid(APP_ID, APP_CERTIFICATE, params.channel, params.uid, role, expireTime, privilegeExpireTime);
+
+return new Response(JSON.stringify({
+    rtcToken: token
+}), { headers })
+```
+
+You now have a fully working token server that you can use to develop your applications. You should use this token server primarily for development purposes. If you are building in production, you should add some user verification and other security measures to ensure that the only people joining calls are the ones who are supposed to.
```

Comment:
- Is this the end? 
- What about how to test the endpoint? How do I know it's working?

Reply:
Added a section called "Run the Backend"

> ## Run the Backend
> Run this token generator using `npm run dev`; now you can type arbitrary values into the URL to receive a token. For example, the URL `localhost:4321/rtc/test/publisher/1.json` means the user with uid `1` requests a token for the `test` channel and wants to publish and receive data. When you type this into the URL you should receive a response of: 
> 
> ```
> {"rtcToken":"<a long string of letters and numbers>"}
> ```